lib/yggdrasil/command/parser_builder.ex

```diff
@@ -0,0 +1,64 @@
+defmodule Yggdrasil.Command.ParserBuilder do
+  alias Yggdrasil.Command.ParseRule
+
+  defmacro __using__(_) do
+    quote do
+      Module.register_attribute __MODULE__, :parse_rules, accumulate: true
+
+      import Yggdrasil.Command.ParserBuilder
+
+      @before_compile unquote(__MODULE__)
+    end
+  end
+
+  defmacro parse(text, command) do
+    quote do
+      @parse_rules ParseRule.create(unquote(text), unquote(command))
```

Comment:
Would you need to downcase here as well? So that text is forced lower

Thinking about this in the parser

parse "GO :exit" va parse "go :exit"